File: examples/go/reaction_role.go
// 反应角色最小示例。
//
//	cd examples/go && go run .  （需设置环境变量）
package main

import (
	"encoding/json"
	"log"
	"os"

	newtbot "github.com/NewtSpeak/NewtBotSdk/go"
)

func main() {
	bot := newtbot.New(os.Getenv("NEWT_BASE_URL"), os.Getenv("NEWT_BOT_TOKEN"))
	rulesMessageID := os.Getenv("RULES_MESSAGE_ID")
	verifiedRoleID := os.Getenv("VERIFIED_ROLE_ID")

	gw := bot.ConnectGateway()
	gw.On("READY", func(payload json.RawMessage) {
		log.Printf("已连接: %s", string(payload))
	})
	gw.On("MESSAGE_REACTION_ADD", func(payload json.RawMessage) {
		var ev struct {
			MessageID string `json:"message_id"`
			GuildID   string `json:"guild_id"`
			UserID    string `json:"user_id"`
			Emoji     string `json:"emoji"`
		}
		if err := json.Unmarshal(payload, &ev); err != nil {
			return
		}
		if ev.MessageID != rulesMessageID || ev.Emoji != "✅" {
			return
		}
		if _, err := bot.AddMemberRole(ev.GuildID, ev.UserID, verifiedRoleID); err != nil {
			log.Printf("赋角失败: %v", err)
			return
		}
		log.Printf("已赋角 %s", ev.UserID)
	})

	select {}
}

File: examples/javascript/reaction-role.mjs
/**
 * 反应角色最小示例。
 *
 *   export NEWT_BASE_URL=https://newt.example.com
 *   export NEWT_BOT_TOKEN=newtbot_xxx
 *   export RULES_MESSAGE_ID=...
 *   export VERIFIED_ROLE_ID=...
 *   node examples/javascript/reaction-role.mjs
 */

import { OwlBotClient } from "../../javascript/index.js"

const bot = new OwlBotClient({
  baseUrl: process.env.NEWT_BASE_URL,
  token: process.env.NEWT_BOT_TOKEN,
})

const RULES_MESSAGE_ID = process.env.RULES_MESSAGE_ID
const VERIFIED_ROLE_ID = process.env.VERIFIED_ROLE_ID

const gw = bot.connectGateway()
gw.on("ready", (data) => console.log("已连接", data?.user?.username))
gw.on("MESSAGE_REACTION_ADD", async (ev) => {
  if (String(ev.message_id) !== RULES_MESSAGE_ID || ev.emoji !== "✅") return
  try {
    await bot.addMemberRole(ev.guild_id, ev.user_id, VERIFIED_ROLE_ID)
    console.log("已赋角", ev.user_id)
  } catch (err) {
    console.error("赋角失败", err)
  }
})
